### ISSUE TITLE
Func parameter inferred types detected as a breaking change

### ISSUE THREAD

**Issue by bradleyfalzon** [bradleyfalzon/abicheck#16]
The following is not a breaking change:

```
go1.5:ftoa.go:50: breaking change parameter types changed
        func AppendFloat(dst []byte, f float64, fmt byte, prec int, bitSize int) []byte
        func AppendFloat(dst []byte, f float64, fmt byte, prec, bitSize int) []byte
```

**Comment by bradleyfalzon** [comment]
Needs to be fixed by normalising the fields so there's just one name per type, similar to a276ea18771545033e3f5152b5eb0e15739103b8

**Comment by bradleyfalzon** [comment]
This could also occur on func and interface types for both parameters and results. Fortunately the related a276ea18771545033e3f5152b5eb0e15739103b8 works on []*ast.Field, which all mentioned types are, so the code there just needs to be abstracted into its own function with a parameter `exportedOnly` which removes variables that aren't exported for the struct cases.

**Comment by bradleyfalzon** [comment]
Fixed in `01b4663788253d32ef559f8bae451d92f6cce9bd`
